test/pylib/scylla_server.py

```diff
@@ -364,6 +364,27 @@ async def stop(self) -> None:
             self.control_connection = None
             self.control_cluster = None
 
+    async def stop_gracefully(self) -> None:
+        """Stop a running server. No-op if not running. Uses SIGTERM to
+        stop, so it is graceful. Waits for the process to exit before return."""
+        # Preserve for logging
+        hostname = self.hostname
+        logging.info("gracefully stopping server at host %s", hostname)
+        if not self.cmd:
+            return
+
+        try:
+            self.cmd.terminate()
+        except ProcessLookupError:
+            pass
+        else:
+            await self.cmd.wait()
```

Comment:
@nyh in general I agree, especially since this has now an exposure in the test itself. there is an overall timeout on running tests in jenkins, so I thought it's not urgent and not requested in my review.

Reply:
This is pre-existing. I'll address later.